aspnetcore/fundamentals/host/platform-specific-configuration.md

```diff
@@ -5,22 +5,450 @@ description: Discover how to enhance an ASP.NET Core app from an external assemb
+
+* Contains a hosting startup class, `ServiceKeyInjection`, which implements `IHostingStartup`. `ServiceKeyInjection` adds a pair of service strings to the app's configuration.
+* Includes a `HostingStartup` attribute.
+
+*HostingStartupPackage/ServiceKeyInjection.cs*:
+
+[!code-csharp[](platform-specific-configuration/samples/3.x/HostingStartupPackage/ServiceKeyInjection.cs?name=snippet1)]
+
+The app's Index page reads and renders the configuration values for the two keys set by the package's hosting startup assembly:
+
+*HostingStartupApp/Pages/Index.cshtml.cs*:
+
+[!code-csharp[](platform-specific-configuration/samples/3.x/HostingStartupApp/Pages/Index.cshtml.cs?name=snippet1&highlight=7-8,13-14)]
+
+### Console app without an entry point
+
+*This approach is only available for .NET Core apps, not .NET Framework.*
+
+A dynamic hosting startup enhancement that doesn't require a compile-time reference for activation can be provided in a console app without an entry point that contains a `HostingStartup` attribute. Publishing the console app produces a hosting startup assembly that can be consumed from the runtime store.
+
+A console app without an entry point is used in this process because:
+
+* A dependencies file is required to consume the hosting startup in the hosting startup assembly. A dependencies file is a runnable app asset that's produced by publishing an app, not a library.
+* A library can't be added directly to the [runtime package store](/dotnet/core/deploying/runtime-store), which requires a runnable project that targets the shared runtime.
+
+In the creation of a dynamic hosting startup:
+
+* A hosting startup assembly is created from the console app without an entry point that:
+  * Includes a class that contains the `IHostingStartup` implementation.
+  * Includes a [HostingStartup](xref:Microsoft.AspNetCore.Hosting.HostingStartupAttribute) attribute to identify the `IHostingStartup` implementation class.
+* The console app is published to obtain the hosting startup's dependencies. A consequence of publishing the console app is that unused dependencies are trimmed from the dependencies file.
+* The dependencies file is modified to set the runtime location of the hosting startup assembly.
+* The hosting startup assembly and its dependencies file is placed into the runtime package store. To discover the hosting startup assembly and its dependencies file, they're listed in a pair of environment variables.
+
+The console app references the [Microsoft.AspNetCore.Hosting.Abstractions](https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/) package:
+
+[!code-xml[](platform-specific-configuration/samples-snapshot/3.x/StartupEnhancement.csproj)]
```

Comment:
The internal review build is showing all 2.1 contents for these snippets, but the actual files you're linking to look correct.

Reply:
That's correct ... the build gives us hard links to the 2.1 version. As long as u can change the version and get the other versions, it should be ok. Idk why it does that. We can ask Scott if you want.

Reply:
I did change to 3.1 and still saw 2.1 snippets.

Reply:
Ah, 3.0 works, it's just 3.1 that's falling back to 2.1.

Reply:
Good 😅 ... yeah, since there's no 3.0 (*yet*), that makes sense.